api/routes/api_sports.py

```diff
@@ -1,20 +1,40 @@
-from flask import Blueprint
-from service.api_logic.sports_logic import get_all_sports
+from flask import Blueprint, request
+from service.api_logic.sports_logic import get_all_sports, get_all_leagues_by_sport
 from database.session import SessionLocal
 from api.routes.scripts import get_error_response
+from api.routes.cache import cache
+from exept.exeptions import DatabaseConnectionError
 
 session = SessionLocal()
 sports_app = Blueprint('sports', __name__)
 
-@sports_app.errorhandler(Exception)
-def handle_exception(e):
-    response = {"error in service": str(e)}
-    return get_error_response(response, 500)
+
+@sports_app.errorhandler(DatabaseConnectionError)
+def handle_db_timeout_error(e):
+    response = {"error in data base": str(e)}
+    return get_error_response(response, 503)
+
 
 @sports_app.route('/all', methods=['GET'])
+@cache.cached(timeout=60*60)
 def get_all_sports_endpoint():
     try:
-        sports = get_all_sports()
+        sports = get_all_sports(session)
+        return sports
+    except Exception as e:
+        response = {"error in service": str(e)}
+        return get_error_response(response, 500)
+
+@sports_app.route('/league', methods=['POST'])
+@cache.cached(timeout=60*60)
+def get_all_leagues_endpoint():
+    try:
+        data = request.get_json()
```

Comment:
I see a lot of places like this, where you parse the incoming data manually.
Use DTOs instead. They will handle all the validation and mapping for you (read what mapping to DTO is in API (ChatGPT?))

I am most familiar with Marshmallow library, which provides that functionality, so you can use it

Reply:
And, of course, this should be done for each endpoint we have   :)

**Quote by Aeschylus: “Only through suffering do we learn”**

Reply:
I have already written DTO models
But they are in the test branch
Otherwise, this PR will be constantly updated and not close never)

Reply:
Nice. (Though I'd like to merge only finished code into develop branch, but let it be)

Then fix other comments (if not already) and I'll approve